### DISCUSSION TITLE
Reduce JS file on NextJs for Seo

### DISCUSSION THREAD

**Discussion by onlineverh** [vercel/next.js#38152] [Show and tell]
i use Next Js and Bootstrap for design my website , after i export and upload on website i checked my website on seoability website i give this error 11Js file
what is the way for reduce my js file to 3 js file

thanks

**Comment by icyJoseph** [comment]
Let's see if I understand this. There's a site, seobility (?), which is saying that you have too many JS files?

> There should be a maximum of 3 files java script and 3 files css. - Taken from their site

That's certainly debatable, and sort of reinforces  my reasoning about why some of these SEO tools are not to be followed blindly.

I went to their own homepage, opened the console and run `document.scripts`, they have 7, I guess they also can't pass that test.

Surely, less JS to download, parse, and execute equals faster pages, but it also equals less features, and interactivity. If your page is pretty much static, one could write in pure HTML and stash all styles into one giant CSS file, but then the amount of features to deliver, let alone engagement, would be severely compromised, I think.

Also less files, usually means, larger files, when the thesis is that we had _too many_ to begin with, which is still problematic, and arguably, worse in certain conditions.
